File: Data Structures & Algorithms/longest-consecutive-sequence/submission-8.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int result = 0;
        Set<Integer> seen = new HashSet<>();

        for(int num: nums){
            seen.add(num);
        }

        for(int i = 0; i < nums.length; i++){
            int current = nums[i];
            int counter = 1;

            
                while(seen.contains(current - 1)){
                    counter++;
                    current = current - 1;
                }
                
            

            result = Math.max(result, counter);
        }

        return result;

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-8.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
         left = [0] * len(nums)
         right = [0] * len(nums)
         result = [0] * len(nums)

         value = 1

         for index, num in enumerate(nums):
            left[index] = value
            value *= nums[index]

        
         value = 1

         for index, num in reversed(list(enumerate(nums))):
            right[index] = value
            value *= nums[index]
        
         for index, value in enumerate(nums):
            result[index] = left[index] * right[index]
        
         return result
